src/Performance/Time/Timer.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Performance\Time;
+
+/**
+ * @internal
+ */
+final class Timer
+{
+    /**
+     * @var float|null
+     */
+    private $microTime;
+
+    public function start()
+    {
+        if (null !== $this->microTime) {
```

Comment:
What about using a guard with an assert library here instead? Here's an example with https://github.com/webmozart/assert:

```php
Assert::null($this->microTime, 'Cannot start a timer that has already been started');
```

Since those are exceptions that are not meant to be catch, I think an assertion library is more concise: in case of error the message will be as comprehensible and easy to spot & report and would remove the need to introduce 2 new exception classes for it

Reply:
agree. This is a subject to discuss for @infection/core whether to include this new lib or not.

I have no objections since this lib has no dependencies and really useful, so I have added it to the PR. If anyone agains - let me know so we can revert it.